src/app/components/mnoe-api/provisioning.svc.coffee

```diff
@@ -134,6 +134,19 @@ angular.module 'mnoEnterpriseAngular'
               deferred.resolve(response)
           )
       )
+
+    @getQuote = (s) ->
+      deferred = $q.defer()
```

Comment:
Same comment as in #517 

I don't think we need to use $q.defer here. And the nested promise is just resolving the response.

> If the returned value is a promise, then the further execution is suspended until it settles. After that, the result of that promise is given to the next .then handler.
-- _https://javascript.info/promise-chaining#returning-promises_

So we could just have:
```coffeescript
@getQuote = (s) ->
  MnoeOrganizations.get().then(
   (response) -> 
     quoteParams = {product_id: s.product.id, product_pricing_id: s.product_pricing?.id, custom_data: s.custom_data, organization_id: response.organization.id}
     MnoeApiSvc.one('organizations', response.organization.id).all('quotes').post(quote: quoteParams)
  )
```
Disclaimer: double check that it works as advertised  ;)

Reply:
This works